### DISCUSSION TITLE
Run-out sensor disabled by default, where to change that?

### DISCUSSION THREAD

**Discussion by gitfve** [mriscoc/Ender3V2S1#1149] [General]
I'am confused about this, Run-out sensor disabled by default

Ender-3 V2/S1 Professional Firmware **20230904**

In the description for the fw you say: Run-out sensor disabled by default. And it is, I tested it!
But in the "**common.json**" you have this:
```
{
    "op": "CustomVal",
    "searchfor": "FIL_RUNOUT_ENABLED_DEFAULT",
    "value": "true"
  },
```

The "**configuration.h**" have this:
**#define FIL_RUNOUT_ENABLED_DEFAULT true** // Enable the sensor on startup. Override with M412 followed by M500.
 
Where have you disabled the filament sensor?

**Comment by mriscoc** [comment]
It was here:
https://github.com/mriscoc/Ender3V2S1/blob/b4d84f733333f72bb85dcf886f17535bb36837cd/Marlin/src/lcd/e3v2/proui/dwin.cpp#L1803

But it is going to be changed in the next version to follow the value of `FIL_RUNOUT_ENABLED_DEFAULT`

**Reply by tmuka** [reply]
This is for the default state of the checkbox enable runout checkbox in the UI, right? @gitfve ,Your runout sensor still works after toggling that back on?  

(Asking because I'm not having success adding mine for the first time, and think I probably broke my 4.2.2 board pull-up)

@mriscoc  thanks for the GREAT firmware!

**Reply by mriscoc** [reply]
That only have effect in the initial state of the runout sensor, it can be enabled later via menu or by g-code.

**Comment by gitfve** [comment]
My printer is a Ender3 S1 PRO. It have runout sensor mounted from the factory.
My Setting: Runout Active: High, Enabled: YES.

I tested it by cutting the filament.
When the filament leave the sensor, nothing is shown, BUT when it have uses the length entered in the setup it activate the Advanced Pause.
Then i load the filament and resumed the print.
All working 100%.

**Reply by tmuka** [reply]
Thanks for the confirmation.

**Reply by mriscoc** [reply]
There is also a visual confirmation in the screen when the filament leaves the runout sensor. 
![image](https://github.com/mriscoc/Ender3V2S1/assets/2745567/ca1fb9f6-d17c-4112-8386-9047fe6a2a23)

**Comment by gexgecko** [comment]
Sounds like this is the cause of https://github.com/mriscoc/Ender3V2S1/issues/1162

**Reply by mriscoc** [reply]
For enable/disable filament run-out sensor, you can use the G-code [M412](https://marlinfw.org/docs/gcode/M412.html) or via menu in Advanced/Filament settings:   
  
![image](https://github.com/mriscoc/Ender3V2S1/assets/2745567/f91121e2-bdf5-42b0-b834-df6cba935d60)

Check also the run-out distance.
